method/application/get.js

```diff
@@ -1,68 +1,83 @@
 const pull = require('pull-stream')
-const { isValid } = require('../../spec/application/application')
+const { isRoot, isUpdate } = require('../../spec/application')
 
 module.exports = function GroupApplicationGet (server) {
   return function groupApplicationGet (applicationId, cb) {
+    server.get({ id: applicationId, private: true }, (err, rootData) => {
+      if (err) return cb(err)
+      if (!isRoot(rootData)) return cb(isRoot.errors)
+
+      fetchUpdates(applicationId, (err, updates) => {
+        if (err) return cb(err)
+
+        var initialState = {
+          comments: [getComment(rootData)],
+          addMember: []
+        }
+        const reduced = updates.reduce((acc, curr) => {
+          acc.comments.push(getComment(curr.value))
+          acc.addMember = [
+            ...acc.addMember,
+            ...getAddMember(curr.value)
+          ]
+
+          return acc
+        }, initialState)
+
+        cb(null, {
+          id: applicationId,
+          applicantId: rootData.author,
+          groupId: rootData.content.groupId,
+          groupAdmins: rootData.content.recps.filter(id => id !== rootData.author),
+          ...reduced
+        })
+      })
+    })
+  }
+
+  function fetchUpdates (applicationId, cb) {
     const query = [
       {
         $filter: {
-          dest: applicationId
+          dest: applicationId,
+          value: {
+            content: {
+              tangles: {
+                application: { root: applicationId }
+              }
+            }
+          }
         }
       }
     ]
 
-    server.get({ id: applicationId, private: true }, (rootErr, rootData) => {
-      if (!isValid(rootData)) {
-        return cb(isValid.errors)
-      }
-      if (rootErr) return cb(rootErr)
-      pull(
-        server.backlinks.read({ query }),
-        pull.filter(m => isValid(m)),
-        pull.collect((err, data) => {
-          if (err) return cb(err)
-          let final = {
-            root: applicationId,
-            applicantId: rootData.author,
-            groupId: rootData.content.groupId,
-            recps: rootData.content.recps
-          }
-          /* If there's only the root message */
-          if (!data || data.length < 1) {
-            const res = {
-              ...final,
-              text: [rootData.content.text.append || '']
-            }
-            return cb(null, res)
-          } else {
-            const reduced = data.reduce((prev, curr) => {
-              /* First message*/
-              if (!prev) {
```

Comment:
hey @luandro I noticed this reduce method was quite complex. Have a look at the `reduce` I replaced it with which uses `initialState` (second argument of reduce) to make it simpler to reason about (in my opinion).